File: training-tracker-client/src/types/sportFolder.ts
export interface SportFolder {
  id: number;
  name: string;
  description: string | null;
  color: string;
  icon: string | null;
  isArchived: boolean;
  sessionCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface UpdateSportFolderRequest {
  name: string;
  description: string | null;
  color: string;
  icon: string | null;
  isArchived: boolean;
}

File: TrainingTracker.Api/Models/SportFolder.cs
namespace TrainingTracker.Api.Models;

public class SportFolder
{
    public int Id { get; set; }

    public string UserId { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public string Color { get; set; } = "#3B82F6";

    public string? Icon { get; set; }

    public bool IsArchived { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}